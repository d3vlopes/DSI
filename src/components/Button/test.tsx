import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Button from '.'

describe('<Button />', () => {
  it('should render the primary style and normal size by default', () => {
    renderWithTheme(<Button>Button</Button>)

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '16.1rem',
      'background-color': '#EE4248',
      color: '#FFFFFF',
    })

    expect(button).toHaveStyleRule('background-color', '#de3f45', {
      modifier: ':hover',
    })
  })

  it('should render the large size ', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '16.6rem',
    })
  })

  it('should render the secodary style ', () => {
    renderWithTheme(<Button styles="secondary">Button</Button>)

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'background-color': '#FFFFFF',
      color: '#EE4248',
    })

    expect(button).toHaveStyleRule('background-color', '#f7f6f6', {
      modifier: ':hover',
    })
  })

  it('should render the secodary style and large size', () => {
    renderWithTheme(
      <Button size="large" styles="secondary">
        Button
      </Button>,
    )

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '16.6rem',
      'background-color': '#FFFFFF',
      color: '#EE4248',
    })
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })
})
