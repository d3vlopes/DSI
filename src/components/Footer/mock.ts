export default [
  {
    src: '/img/platform/microsoft.png',
    alt: 'Microsoft Silverlight',
  },
  {
    src: '/img/platform/net.png',
    alt: 'Microsoft .Net Framework',
  },
  {
    src: '/img/platform/java.png',
    alt: 'Java',
  },
  {
    src: '/img/platform/ios.png',
    alt: 'IOS',
  },
  {
    src: '/img/platform/android.png',
    alt: 'Android',
  },
  {
    src: '/img/platform/black-berry.png',
    alt: 'Black Berry',
  },
]
