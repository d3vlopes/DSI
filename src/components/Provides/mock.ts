import { ProvideProps } from '.'

const ProvidesMock: ProvideProps[] = [
  {
    image: 'img/products/product.png',
    title: 'Our Product',
    description:
      'Our product is made on the base of our team’s careful research and analyses, ranging from internet based application.',
  },
  {
    image: 'img/products/service.png',
    title: 'Our Service',
    description:
      'DSI’s shared service solutions focus on the front-end based software development, designed specifically for the banking and financial sectors.',
  },
  {
    image: 'img/products/technology.png',
    title: 'Our Technology',
    description:
      'First JAVA, runs on more than 850 million personal computers worldwide, and on billions of devices worldwide, including mobile and TV devices.',
  },
]

export default ProvidesMock
