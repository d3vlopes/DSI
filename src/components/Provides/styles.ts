import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;

  ${media.greaterThan('medium')`
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
  `}

  ${media.greaterThan('large')`
    flex-flow: row no-wrap;
    justify-content: space-between;
    margin-left: -1.5rem;
  `}
`

export const Provide = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    &:not(:last-of-type) {
      margin-bottom: ${theme.spacings.xlarge};
    }

    ${media.greaterThan('medium')`
      margin-left: 5.3rem;
    `}

    ${media.greaterThan('large')`
      align-items: flex-start;
      margin-left: 5.3rem;
      justify-content: initial;
    `}

    > img {
      max-width: 100%;
    }

    > h3 {
      font-size: ${theme.font.sizes.small};
      font-weight: ${theme.font.weight.medium};
      line-height: 133.69%;
      color: ${theme.colors.primary};
      margin: ${theme.spacings.small} 0;
    }

    > p {
      max-width: 35.9rem;
      line-height: 150%;
      text-align: center;
      color: ${theme.colors.black};
      margin: ${theme.spacings.small} 0;

      ${media.greaterThan('large')`
        text-align: initial;
        max-width: 31.9rem;
      `}
    }
  `}
`
