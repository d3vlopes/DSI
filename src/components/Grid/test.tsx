import { renderWithTheme } from 'utils/test/helpers'

import { Grid } from '.'

describe('<Grid />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Grid>Children</Grid>)

    expect(container.firstChild).toMatchInlineSnapshot(`
      .c0 {
        display: grid;
        grid-template-columns: repeat(auto-fill,minmax(23rem,1fr));
        grid-gap: 2.4rem;
        margin: 2.4rem 0;
      }

      <div
        class="c0"
      >
        Children
      </div>
    `)
  })
})
