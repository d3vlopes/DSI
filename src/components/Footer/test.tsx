import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render Footer', () => {
    renderWithTheme(<Footer />)

    expect(screen.getByRole('heading', { name: /Footer/i })).toBeInTheDocument()
  })
})
