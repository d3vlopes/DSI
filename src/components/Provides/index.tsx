import Button from 'components/Button'

import * as S from './styles'

export type ProvideProps = {
  image: string
  title: string
  description: string
}

export type ProvidesProps = {
  items: ProvideProps[]
}

const Provides = ({ items }: ProvidesProps) => (
  <S.Wrapper>
    {items?.map((item, index) => (
      <S.Provide key={`provide-${index}`} id="provide">
        <img src={`${item.image}`} alt={`${item.title}`} />
        <h3>{item.title}</h3>
        <p>{item.description}</p>
        <Button size="small">Read More</Button>
      </S.Provide>
    ))}
  </S.Wrapper>
)

export default Provides
