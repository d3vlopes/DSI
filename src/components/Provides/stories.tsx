import { Story, Meta } from '@storybook/react/types-6-0'

import Provides, { ProvidesProps } from '.'
import items from './mock'

export default {
  title: 'Provides',
  component: Provides,
  args: {
    items,
  },
} as Meta

export const Default: Story<ProvidesProps> = (args) => (
  <div style={{ maxWidth: '130rem', margin: '0 auto' }}>
    <Provides {...args} />
  </div>
)
