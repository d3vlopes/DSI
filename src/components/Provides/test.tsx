import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Provides from '.'
import mock from './mock'

describe('<Provides />', () => {
  it('should render only one Provide', () => {
    renderWithTheme(<Provides items={mock.slice(0, 1)} />)

    expect(screen.getByRole('img', { name: mock[0].title })).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: mock[0].title }),
    ).toBeInTheDocument()

    expect(screen.getByText(mock[0].description)).toBeInTheDocument()

    expect(
      screen.getByRole('button', { name: /read more/i }),
    ).toBeInTheDocument()
  })

  it('should render three Provides', () => {
    const { container } = renderWithTheme(<Provides items={mock} />)

    expect(container.querySelectorAll('#provide')).toHaveLength(3)
    expect(container.firstChild).toMatchSnapshot()
  })
})
