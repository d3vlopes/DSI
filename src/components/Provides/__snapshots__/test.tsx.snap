// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Provides /> should render three Provides 1`] = `
.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 1.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 2.0rem;
  box-shadow: 1px 2px 4px rgba(0,0,0,0.25);
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  width: 11.9rem;
  height: 4.2rem;
  font-size: 1.8rem;
  line-height: 144.69%;
  background-color: #EE4248;
  color: #FFFFFF;
}

.c2:hover {
  background-color: #de3f45;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1:not(:last-of-type) {
  margin-bottom: 3.2rem;
}

.c1 > img {
  max-width: 100%;
}

.c1 > h3 {
  font-size: 2.8rem;
  font-weight: 500;
  line-height: 133.69%;
  color: #EE4248;
  margin: 1.6rem 0;
}

.c1 > p {
  max-width: 35.9rem;
  line-height: 150%;
  text-align: center;
  color: #30444E;
  margin: 1.6rem 0;
}

@media (min-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-flow: row wrap;
    -ms-flex-flow: row wrap;
    flex-flow: row wrap;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

@media (min-width:1170px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:768px) {
  .c1 {
    margin-left: 5.3rem;
  }
}

@media (min-width:1170px) {
  .c1 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    margin-left: 5.3rem;
  }
}

@media (min-width:1170px) {
  .c1 > p {
    text-align: initial;
    max-width: 31.9rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    id="provide"
  >
    <img
      alt="Our Product"
      src="img/products/product.png"
    />
    <h3>
      Our Product
    </h3>
    <p>
      Our product is made on the base of our team’s careful research and analyses, ranging from internet based application.
    </p>
    <button
      class="c2"
    >
      <span>
        Read More
      </span>
    </button>
  </div>
  <div
    class="c1"
    id="provide"
  >
    <img
      alt="Our Service"
      src="img/products/service.png"
    />
    <h3>
      Our Service
    </h3>
    <p>
      DSI’s shared service solutions focus on the front-end based software development, designed specifically for the banking and financial sectors.
    </p>
    <button
      class="c2"
    >
      <span>
        Read More
      </span>
    </button>
  </div>
  <div
    class="c1"
    id="provide"
  >
    <img
      alt="Our Technology"
      src="img/products/technology.png"
    />
    <h3>
      Our Technology
    </h3>
    <p>
      First JAVA, runs on more than 850 million personal computers worldwide, and on billions of devices worldwide, including mobile and TV devices.
    </p>
    <button
      class="c2"
    >
      <span>
        Read More
      </span>
    </button>
  </div>
</div>
`;
